File: app/src/main/java/com/wxm/camerajob/ui/acutility/ACJobSlide.java
package com.wxm.camerajob.ui.acutility;

import android.content.Intent;
import com.wxm.camerajob.base.data.GlobalDef;
import com.wxm.camerajob.ui.fragment.utility.FrgJobSlide;

import java.util.LinkedList;

import cn.wxm.andriodutillib.ExActivity.BaseAppCompatActivity;
import cn.wxm.andriodutillib.util.FileUtil;
import cn.wxm.andriodutillib.util.UtilFun;

/**
 * 幻灯片展示Job
 */
public class ACJobSlide extends BaseAppCompatActivity {

    @Override
    protected void leaveActivity() {
        int ret_data = GlobalDef.INTRET_USR_LOGOUT;

        Intent data = new Intent();
        setResult(ret_data, data);
        finish();
    }

    @Override
    protected void initFrgHolder() {
        LOG_TAG = "ACJobSlide";

        Intent it = getIntent();
        if (null == it)
            return;

        String sp = it.getStringExtra(GlobalDef.STR_LOAD_PHOTODIR);
        if (UtilFun.StringIsNullOrEmpty(sp))
            return;

        LinkedList<String> ll_photo = FileUtil.getDirFiles(sp, "jpg", false);
        if (UtilFun.ListIsNullOrEmpty(ll_photo))
            return;

        mFGHolder = FrgJobSlide.newInstance(ll_photo);
    }
}

File: app/src/main/java/com/wxm/camerajob/ui/Job/slide/ACJobSlide.kt
package com.wxm.camerajob.ui.Job.slide

import android.content.Intent
import android.os.Bundle
import com.wxm.camerajob.data.define.EAction
import com.wxm.camerajob.data.define.GlobalDef
import wxm.androidutil.Switcher.ACSwitcherActivity
import wxm.androidutil.util.FileUtil

/**
 * slide show ui for job
 */
class ACJobSlide : ACSwitcherActivity<FrgJobSlide>() {
    override fun leaveActivity() {
        setResult(GlobalDef.INTRET_USR_LOGOUT, Intent())
        finish()
    }

    override fun setupFragment(p0: Bundle?) {
        intent?.let {
            it.getStringExtra(EAction.LOAD_PHOTO_DIR.actName)?.let {
                FileUtil.getDirFiles(it, "jpg", false)?.let {
                    addFragment(FrgJobSlide.newInstance(it))
                }
            }
        }
    }
}


